src/Compiler/Service/TransparentCompiler.fs

```diff
@@ -1575,6 +1574,39 @@ type internal TransparentCompiler
             }
         )
 
+    let TryGetRecentCheckResultsForFile
+        (
+            fileName: string,
+            projectSnapshot: FSharpProjectSnapshot,
+            userOpName: string
+        ) : Async<(FSharpParseFileResults * FSharpCheckFileResults * SourceTextHash) option> =
+        ignore userOpName
+
+        let file =
+            projectSnapshot.ProjectSnapshot.SourceFiles
+            |> List.tryFind (fun f -> f.FileName = fileName)
+
+        match file with
+        | Some f ->
+            async {
+                let! source = f.GetSource() |> Async.AwaitTask
+                let sourceHash = source.GetHashCode() |> int64
+
+                let cacheKey =
+                    projectSnapshot.ProjectSnapshot.FileKeyWithExtraFileSnapshotVersion fileName
+
+                let version = cacheKey.GetVersion()
+
+                let parseFileResultsAndcheckFileAnswer =
+                    caches.ParseAndCheckFileInProject.TryGet(cacheKey.GetKey(), (fun cachedVersion -> cachedVersion = version))
```

Comment:
With this I'd expect the tests to be failing since the `version` here should contain all the previous files' versions. If they pass then something's wrong somewhere :)

Reply:
Can you elaborate a bit for me? As far as I understand, we are also caching with that version in `ComputeParseAndCheckFileInProject`

Reply:
This version should change any time any prerequisites for the file checking change - so that makes sense for `ComputeParseAndCheckFileInProject` because we need to check it again. 

But here we want to return a result if we have it for this file with the same _file version_ = contents of this file didn't change.

So if a file above changes, `cacheKey.GetVersion()` will return something else and we won't have a result for it. That's why I would expect some of the tests to fail. 

So I would expect here `(fun (_fullVersion, fileContentVersion) -> fileContentVersion = (snd version))`. Like we had when we were comparing the file hash. That's why we added the file's version separately. If we don't use it, we don't need to add it. Also we wouldn't need the predicate function, since we could just directly look up the key + version like we normally do. 

Does it make sense?

Reply:
Ah yeah, okay. Makes total sense. Thanks for your patience!